boilerplate/app/components/Toggle.tsx

```diff
@@ -0,0 +1,493 @@
+    </Wrapper>
+  )
+}
+
+const ToggleInputs: Record<Variants, FC<ToggleInputProps>> = {
+  checkbox: Checkbox,
+  switch: Switch,
+  radio: Radio,
+}
+
+function Checkbox(props: ToggleInputProps) {
+  const {
+    on,
+    status,
+    disabled,
+    innerStyle: $checkboxContainerStyleOverride,
+    detailStyle: $checkboxIconStyleOverride,
+  } = props
+
+  const $animatedCheckboxContainer = useAnimatedStyle(
+    () => ({ opacity: withTiming(on ? 1 : 0) }),
+    [on],
+  )
+
+  const $checkboxContainerStyle = [
+    $checkboxContainer,
+    { backgroundColor: colors.palette.secondary500 },
+    status === "error" && { backgroundColor: colors.error },
+    disabled && { backgroundColor: colors.transparent },
+    $animatedCheckboxContainer,
+    $checkboxContainerStyleOverride,
+  ]
+
+  const $checkboxIconStyle = [
+    $checkboxIcon,
+    { tintColor: colors.palette.neutral100 },
+    disabled && { tintColor: colors.palette.neutral600 },
+    $checkboxIconStyleOverride,
+  ]
+
+  return (
+    <Animated.View style={$checkboxContainerStyle}>
+      <Image source={iconRegistry.check} style={$checkboxIconStyle} />
+    </Animated.View>
+  )
+}
+
+function Radio(props: ToggleInputProps) {
+  const {
+    on,
+    status,
+    disabled,
+    innerStyle: $radioContainerStyleOverride,
+    detailStyle: $radioDotStyleOverride,
+  } = props
+
+  const $animatedRadioContainer = useAnimatedStyle(
+    () => ({ opacity: withTiming(on ? 1 : 0) }),
+    [on],
+  )
+
+  const $radioContainerStyle = [
+    $radioContainer,
+    { backgroundColor: colors.palette.neutral100 },
+    disabled && { backgroundColor: colors.transparent },
+    $animatedRadioContainer,
+    $radioContainerStyleOverride,
+  ]
+  const $radioDotStyle = [
+    $radioDot,
+    { backgroundColor: colors.palette.secondary500 },
+    status === "error" && { backgroundColor: colors.error },
+    disabled && { backgroundColor: colors.palette.neutral600 },
+    $radioDotStyleOverride,
+  ]
+
+  return (
+    <Animated.View style={$radioContainerStyle}>
+      <View style={$radioDotStyle} />
+    </Animated.View>
+  )
+}
+
+function Switch(props: ToggleInputProps) {
+  const {
+    on,
+    status,
+    disabled,
+    innerStyle: $switchContainerStyleOverride,
+    detailStyle: $switchKnobStyleOverride,
+  } = props
+  const [renderedKnobWidth, setRenderedKnobWidth] = useState(0)
+
+  const $animatedSwitchContainer = useAnimatedStyle(() => {
+    const onBackgroundColor = (function () {
+      if ($switchContainerStyleOverride?.backgroundColor) {
+        return $switchContainerStyleOverride.backgroundColor
+      }
+      if (disabled) return colors.transparent
+      if (status === "error") return colors.error
+      return colors.palette.secondary500
+    })()
+
+    const backgroundColor = withTiming(on ? onBackgroundColor : colors.transparent)
+
+    return { backgroundColor }
+  }, [status, disabled, on])
```

Comment:
The animation for the switches track. Since we can't simply fade in/out the track (due to a nested knob), we animate the background color instead. This could actually be done with a fade, but we'd have to un-nest the knob. I will experiment with this. Interpolating colors isn't ideal in certain situations since there isn't any control over the tone/hue when transitioning - with some color customizations, you might see a weird color transition.